### DISCUSSION TITLE
Aglolia Refinement List With Titles - Empty Refinement Lists

### DISCUSSION THREAD

**Discussion by JSShant** [algolia/instantsearch#5664] [Q&A]
![image](https://github.com/algolia/instantsearch/assets/105369293/dabbc5c4-61d2-4fa6-8e2a-5c501f57a77e)

We have facets to display on certain pages, I would like to create a title for each refinementList container however attributes to render always returns the big array of all facets meaning there are some titles which show up without any values like the image below:
![image](https://github.com/algolia/instantsearch/assets/105369293/1efe6a1d-5a3a-4c87-aa2b-0501f9d92ea9)
How can I conditionally render this component based on whether there are values in refinementlist or not? Am planning to eventually make this into a collapsible component. Thankyou

**Comment by Haroenv** [comment]
If you're ok with customising the look of the refinement list itself (which seems to be the case), you can use the connector to make a custom refinement list which can collapse and be hidden if there are no results:

```jsx
function RefinementList(props) {
  const { items } = useRefinementList(props, { $$widgetType: 'cmty.collapsibleRefinementList` })
 
  if (!items.length) return;

  return <>{/* ... rendering of the list */}</>
}
```

**Reply by JSShant** [reply]
Hmm thanks for this, to my understanding though if i were to do this, the refinement list may not show but the titles will definitely show as attributes to render will still return. Is there a way I can create a custom component with both useDynamicWidgets() and useRefinementList()
and render the whole component based on items.length? Maybe this component could be used as the fallback component as well, not 100%

**Reply by Haroenv** [reply]
Sorry, I was unclear but I meant that the title also goes in the "rendering of the list" part, so that it shows together with the values of the list itself.
